### PR TITLE
Add in optional application query param for login endpoint

### LINKED ISSUE #4440: Parameters to identify invoking application and state
When My VA Health (Cerner) triggers the standalone login page, they need to at a minimum pass a parameter that identifies the application. Additionally they may desire a state/relaystate type parameter that can be echoed back so that they can return user to any specific state/context that they were in before redirecting.

Assumption: the base callback URL will be statically configured for each application (i.e. we won't allow a dynamic callback URL to be presented as a separate parameter). 

Unknown: Do we want to maintain this mapping from application to callback URL in frontend code or in vets-api?
Unknown: What's needed to maintain the relaystate presented. by the My VA Health (Cerner) application across. all of the multiple SAML requests that we will be making to authenticate the user.

### PR DESCRIPTION
**Pull request by hinzed1127**
## Description

Closes  department-of-veterans-affairs/va.gov-team#4440

Client-side changes to complement department-of-veterans-affairs/vets-api#4042 . This PR adds in an `application` param to the login utility function. This is so we can add an application query param to the new sessions endpoint, to redirect to an external application upon successful authentication (redirect happens via `vets-api`).

## Testing done

Manual verification with up-to-date changes from `vets-api

## Acceptance criteria
- [ ] hitting `/sessions/{type}/new?application=myvahealth` results in the expected redirect

## Definition of done
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/login/containers/SignInApp.jsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ import downtimeBanners from '../utilities/downtimeBanners';
 const loginHandler = loginType => () => {
   // TODO add separate login tracking for this page
   //   recordEvent({ event: `login-attempted-${loginType}` });
-  login(loginType, 'v1');
+  login(loginType, 'v1', 'cerner');
```
Comment (ericbuckley) [inline src/applications/login/containers/SignInApp.jsx]: does this mean that everyone landing on the standalone login page will be redirected back to cerner?

Comment (ericbuckley) [reply src/applications/login/containers/SignInApp.jsx]: also this is getting changed to "myvahealth" https://github.com/department-of-veterans-affairs/vets-api/pull/4055

Comment (hinzed1127) [reply src/applications/login/containers/SignInApp.jsx]: @ericbuckley I'll update the parameter, as well as refactor it to expect the same query param that the api expects.

ie, `/sign-in?application=myvahealth` will pass along the same query to the sessions endpoint, where other users will just make a normal sessions request.

Comment (hinzed1127) [reply src/applications/login/containers/SignInApp.jsx]: along the lines of avoiding bad side-effects if non-cerner users were to navigate to this page, should the same feature flagging logic that the sign-in modal needs get added here? 

I.e., if someone navigates to this page not from cerner **and** not feature flagged for SSO, should the standalone page also hit the `v0` sessions endpoint?

Comment (ericbuckley) [reply src/applications/login/containers/SignInApp.jsx]: I would think no.  We're not purposely directing users to that page, so while it should work, I don't think we need to guarantee the same experience they would get when logging in via the modal on the home page.

Comment (patrickvinograd) [reply src/applications/login/containers/SignInApp.jsx]: Agreed, I don't think we need to feature flag the standalone login page, it only makes sense in SSO scenarios

**Review by patrickvinograd — COMMENTED** [review]
This looks on  the right track as soon as Eric's comments are accounted for.

File: src/applications/login/containers/SignInApp.jsx [hunk 2]
```diff
@@ -59,7 +55,9 @@ class SignInPage extends React.Component {
 
   render() {
     const { globalDowntime } = this.state;
-    const loggedOut = this.props.location.query.auth === 'logged_out';
+    const { query } = this.props.location;
+    const loggedOut = query.auth === 'logged_out';
+    const application = query.application;
```
Comment (hinzed1127) [inline src/applications/login/containers/SignInApp.jsx]: this gives us flexibility to direct to other applications or if the `myvahealth` param were to change, too

**Review by ericbuckley — APPROVED** [review]
LGTM

**Review by rianfowler — APPROVED** [review]

**Pull request merged**